system-int-test/test/cmr/system_int_test/search/variable/variable_association_test.clj

```diff
@@ -0,0 +1,428 @@
+           (dissociate-variable-fn token "variable100" request-json))
+        ; vu/dissociate-by-query {:provider "foo"}
+        vu/dissociate-by-concept-ids [{:concept-id coll-concept-id}]))
+
+    ;; This test is commented out since we don't support delete variable yet
+    ;; Once we support delete variable, re-enable this test.
+    #_(testing "Dissociate deleted variable"
+        (vu/delete-variable token variable-name)
+        (are [dissociate-variable-fn request-json]
+          (= {:status 404
+              :errors [(format "Variable with variable-name [%s] was deleted." variable-name)]}
+             (dissociate-variable-fn token variable-name request-json))
+          ; vu/dissociate-by-query {:provider "foo"}
+          vu/dissociate-by-concept-ids [{:concept-id coll-concept-id}]))))
+
+(deftest dissociate-variables-with-mixed-response-test
+  (e/grant-registered-users (s/context) (e/coll-catalog-item-id "PROV1"))
+  (testing "dissociate variable with mixed success and failure response"
+    (let [coll1 (d/ingest "PROV1" (dc/collection {:entry-title "ET1"}))
+          coll2 (d/ingest "PROV1" (dc/collection {:entry-title "ET2"}))
+          coll3 (d/ingest "PROV1" (dc/collection {:entry-title "ET3"}))
+          token (e/login (s/context) "user1")
+          variable-name "variable1"
+          assert-variable-associated (partial vu/assert-variable-associated-with-query
+                                              token {:variable-name "variable1"})]
+      (vu/create-variable token (vu/make-variable {:Name variable-name}))
+      (index/wait-until-indexed)
+      (vu/associate-by-concept-ids token variable-name [{:concept-id (:concept-id coll1)}
+                                                        {:concept-id (:concept-id coll2)
+                                                         :revision-id (:revision-id coll2)}])
+      (assert-variable-associated [coll1 coll2])
+
+      (let [response (vu/dissociate-by-concept-ids
+                      token variable-name
+                      [{:concept-id "C100-P5"} ;; non-existent collection
+                       {:concept-id (:concept-id coll1)} ;; success
+                       {:concept-id (:concept-id coll2) :revision-id 1} ;; success
+                       {:concept-id (:concept-id coll3)}])] ;; no variable association
+
+        (vu/assert-variable-dissociation-response-ok?
+         {["C100-P5"] {:errors ["Collection [C100-P5] does not exist or is not visible."]}
+          ["C1200000012-PROV1"] {:concept-id "VA1200000016-CMR" :revision-id 2}
+          ["C1200000013-PROV1" 1] {:concept-id "VA1200000017-CMR" :revision-id 2}
+          ["C1200000014-PROV1"] {:warnings ["Variable [variable1] is not associated with collection [C1200000014-PROV1]."]}}
+         response)
+        (assert-variable-associated [])))))
+
+; ;; This tests association retention when collections and variables are updated or deleted.
```

Comment:
Similar add a reason for why these are commented out. Write a ticket if there is nothing already written to track adding this capability.